src/test/integ/actions.test.ts

```diff
@@ -64,7 +64,13 @@ describe("Eval VSCode", () => {
         output = await eval_from_nvim(client, "function foo() {}; return foo;");
         assert.equal(output, null);
 
-        await assert.rejects(() => eval_from_nvim(client, "const a = {}; a.a = a; return a"));
+        output = await eval_from_nvim(client, "function foo() {}; return {a: foo, b: 123};");
+        assert.deepEqual(output, { b: 123 });
+
+        await assert.rejects(
+            () => eval_from_nvim(client, "const a = {}; a.a = a; return a"),
+            /Error executing lua Return value of eval not JSON serializable: TypeError: Converting circular structure to JSON.*/,
```

Comment:
A bit of a nitpick, but should this map so exactly to the specific TypeError? Maybe just change this to `Error executing lua Return value of eval not JSON serializable.*`, since really that's what we care about?

Reply:
my thinking here is that we are testing that the user is given information about why the serialisation failed, but I can make the adjustment if you prefer.

Reply:
Nah, don't sweat it :) it's a taste thing. I don't know that we need to guarantee such implementation details. No biggie either way